### PR TITLE
New group for app level testing and misc refactoring

### PR DESCRIPTION
**Pull request by webteckie**
## Description of changes

This change set includes the introduction of a new group for application level testing and uses the latest app page object api to carry on the testing.  The change set also adds support for a front page and its testing.  Since the app PO now supports automatic waiting for the next target element, manual waiting was eliminated in many places thus simplifying the tests.

## Related issues (if any)

N/A

## Testing

- [X] Please confirm `npm run test-all` ran successfully.

### CODE REVIEW

File: test/e2e/adminUI/tests/group003Home/uiTestHomeView.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+		browser.adminUISigninScreen.signin();
+	},
+	after: function (browser) {
+		browser.adminUIApp.signout();
+		browser.end();
+	},
+	'Home view should have a dashboard header': function (browser) {
+		browser.adminUIHomeScreen.expect.element('@dashboardHeader')
+			.text.to.equal('e2e');
```
Comment (jstockwin) [inline test/e2e/adminUI/tests/group003Home/uiTestHomeView.js]: This looks like it's still in need of refactoring, is that coming later?

File: test/e2e/server.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 var async = require('async');
 var keystone = require('../..');
 var ReactEngine = require('react-engine');
-var view = require('react-engine/lib/expressView');
+//var view = require('react-engine/lib/expressView');
```
Comment (jstockwin) [inline test/e2e/server.js]: Remove this line entirely?

**Comment by jstockwin** [comment]
@webteckie Not sure I understand the need for testing a front page, but if you want it in then fair enough. 

There's a commented out line that needs removing I think (see inline comments). 

Just re-running the testing now that I've published the new keystone-nightwatch-e2e version.

**Comment by webteckie** [comment]
> Not sure I understand the need for testing a front page, but if you want it in then fair enough.

@jstockwin the admin app has a front page button.  Need to test that somehow :-)

**Pull request merged**